bazel/linux/templates/init.template.sh

```diff
@@ -1,7 +1,8 @@
 #!/bin/sh
 
 echo ========= {message} - {target} ==========
-trap "poweroff -f" EXIT
+# Power off the VM in case of ERR; prevent the next script from running.
+trap "poweroff -f" ERR
```

Comment:
It should work with UML as well. I haven't tested, though. Any good ideas how should I call it? `Power off the VM/UML`? Or `Abort in case of error`?

Reply:
uml behaves like a VM, must call poweroff.

There are two ways to exit VM/QEMU:
1) The script just exits - what happens here is the kernel panics, prints a stack trace, and complains that init died.
2) The script calls poweroff - the kernel asks the emulator to exit, all goes down cleanly.

So, why do we need poweroff only in case of error? If we want to avoid poweroff when doing things manually, we probably need some other strategy (eg, is the script running as PID 1? was it started from another shell? ...)

Reply:
It goes like that:
* `ib/apps/sf_111_in_vm-init.sh` starts
  * runs `ib/apps/qemu_kunit_test-init.sh`
    * exits so poweroff
  * it should run `ib/apps/sf_111_test` but never reaches here

Both of the `*init.sh` have this line `trap "poweroff -f" EXIT` so my actual test script never runs.

`ib/apps/sf_111_in_vm-init.sh` contents are:
```
echo '==== run: //driver/ib:setup-modules as "driver/ib/setup-modules "...'
'driver/ib/setup-modules'
echo '==== run: //ib/apps:qemu_kunit_test as "ib/apps/qemu_kunit_test-init.sh "...'
'ib/apps/qemu_kunit_test-init.sh'
echo '==== run: //ib/apps:sf_111_test as "ib/apps/sf_111_test "...'
'ib/apps/sf_111_test'
```

Any ideas how to fix that?

Reply:
oh, in that case I'd just two separate templates: one for the kunit test, one for the overall running script. Before, kunit was self-standing, now it's just another test. No individual test should be shutting down the machine.

The overall "run all tests in turn" should exit once the first test fails. I remember having a `set -e` in there, but would be good to test.